src/udev/udevd.c

```diff
@@ -1010,6 +973,36 @@ static int event_queue_start(Manager *manager) {
                 if (r > 0)
                         continue;
 
+                r = event_lock_block_device(event);
```

Comment:
so this will open the block device from the main udev process, and i think that's a problem. opening the device is a device access, and that might be slow/blocking. i.e. think about floppies and such. It's ok to open devices from worker processes, because that's what they are for: parallelize potentially slow stuff. But if the main process opens this and stalls then all of udev is just hanging, and that#s not OK.

i think it's essential that opening the device remains in the worker hence.

Reply:
Hmm, indeed...